testutil/db.go

```diff
@@ -204,3 +204,37 @@ func NewBlockAdapterByEnv(t *testing.T, translator block.UploadIDTranslator) blo
 		return mem.New(mem.WithTranslator(translator))
 	}
 }
+
+func NewBlockAdapterFactory(t *testing.T, translator block.UploadIDTranslator, blockstoreType string) block.Adapter {
+	switch blockstoreType {
+	case gs.BlockstoreType:
+		ctx := context.Background()
+		client, err := storage.NewClient(ctx)
+		if err != nil {
+			t.Fatal("Google Storage new client", err)
+		}
+		return gs.NewAdapter(client, gs.WithTranslator(translator))
+
+	case lakefsS3.BlockstoreType:
+		awsRegion, regionOk := os.LookupEnv(envKeyAwsRegion)
```

Comment:
Not sure I understand.  `config` is exactly a mechanism to allow configuring everything from defaults or a config file or environment variables.  This code seems to bypass that (and `BuildBlockAdapter`) in favour of what looks like _another_ mechanism to allow configuring everything from defaults or environment variables.  Could we use `config` and the existing config, possibly just changing some default values for testing?

Reply:
I think the main issue here is that this code is almost a duplication of the function above it, `NewBlockAdapterByEnv` (which may suffer from what @arielshaqed is saying but it's out of the scope of this PR to fix it).

@eylonronen, instead I think `NewBlockAdapterByEnv` should be changed to `NewBlockAdapterByType`. Its callers will extract the type from the env variable (the function will still lookup some other stuff in the environment variables like `envKeyAwsSecretKey` and that is ok for now).

Reply:
That would also work for me, thanks!